backend/algorithms/objects/conditions.py

```diff
@@ -239,16 +254,17 @@ def _validate_course(category: Category):
                 unlocked, warnings = list(zip(*validations))
                 satisfied = all(unlocked) if category.logic == Logic.AND else any(unlocked)
                 return satisfied, warnings
+
             elif isinstance(category, ClassCategory):
                 course = category.class_name
                 if course not in user.courses:
-                    return False, []
+                    return False, [f"Need {self.grade} in {course} for this course"]
 
                 user_grade = user.get_grade(course)
                 if user_grade is None:
                     return True, [self.get_warning()]
                 if user_grade < self.grade:
-                    return False, []
+                    return False, [f"Your grade {user_grade} in course {course} does not meet the grade requirements (minimum {self.grade}) for this course"]
                 return True, []
             else:
                 return True, ["We have failed to parse this correctly"]
```

Comment:
*theoretically* it would be funny if u made all of this into a one-line ternary
(probably don't do this) (but if you do, don't say it was my idea)